userspace/engine/rules.cpp

```diff
@@ -195,7 +197,22 @@ int falco_rules::enable_rule(lua_State *ls)
 	std::string rule = rulec;
 	bool enabled = (lua_tonumber(ls, -1) ? true : false);
 
-	rules->enable_rule(rule, enabled);
+	// Escape any regex special characters in the rule name
+	std::string sanitized;
+
+	try {
+		std::regex escape_chars { R"([$\.*+?()[\]{}|^])" };
+
+		sanitized = std::regex_replace(rule, escape_chars, R"(\$&)" );
```

Comment:
```suggestion
		sanitized = std::regex_replace(rule, escape_chars, "$&");
```

Reply:
Thanks, moot now that I switched to find_first_of.